kfs-core/src/main/java/edu/arizona/kfs/sys/batch/service/impl/BuildingImportServiceImpl.java

```diff
@@ -220,75 +218,58 @@ private boolean isValid(ArchibusBuildings archBuilding, PrintWriter outReportErr
-			reportMsg += "BuildingAddressZipCode(" + archBuilding.getBuildingAddressZipCode() + ") is Not Valid, ";;
+		Integer maxBuildingAddressCountryCodeLen = ddService.getAttributeMaxLength(Building.class, "buildingAddressCountryCode");
+
+		if (!isAttributeValid(isBuildingCodeRequired, archBuilding.getBuildingCode(), archBuilding.getBuildingCode().length(), maxBuildingCodeLen, reportMsg, outReportErrorWriter, "BuildingCode(") || 
+			!isAttributeValid(isBuildingNameRequired, archBuilding.getBuildingName(), archBuilding.getBuildingName().length(), maxBuildingNameLen, reportMsg, outReportErrorWriter, "BuildingName(") || 
+			!isAttributeValid(isBuildingStreetAddressRequired, archBuilding.getBuildingStreetAddress(), archBuilding.getBuildingStreetAddress().length(), maxBuildingStreetAddressLen, reportMsg, outReportErrorWriter, "BuildingStreetAddress(") || 
+			!isAttributeValid(isBuildingAddressCityNameRequired, archBuilding.getBuildingAddressCityName(), archBuilding.getBuildingAddressCityName().length(), maxBuildingAddressCityNameLen, reportMsg, outReportErrorWriter, "BuildingAddressCityName(") || 
+			!isAttributeValid(isBuildingAddressStateCodeRequired, archBuilding.getBuildingAddressStateCode(), archBuilding.getBuildingAddressStateCode().length(), maxBuildingAddressStateCodeLen, reportMsg, outReportErrorWriter, "BuildingAddressStateCode(") || 
+			!isAttributeValid(isBuildingAddressZipCodeRequired, archBuilding.getBuildingAddressZipCode(), archBuilding.getBuildingAddressZipCode().length(), maxBuildingAddressZipCodeLen, reportMsg, outReportErrorWriter, "BuildingAddressZipCode(") || 
+			!isAttributeValid(isBuildingAddressCountryCodeRequired, archBuilding.getBuildingAddressCountryCode(), archBuilding.getBuildingAddressCountryCode().length(), maxBuildingAddressCountryCodeLen, reportMsg, outReportErrorWriter, "BuildingAddressCountryCode(")) {
+
+			return false;
 		}
-		if (isAttributeLenToLong(maxBuildingAddressZipCodeLen, archBuilding.getBuildingAddressZipCode().length())) {
+		return true;
+	}
+
+	private Boolean isAttributeValid(Boolean isAttributeRequired, String attributeValue, Integer attributeLen, Integer maxAttributeLen, String reportMsg, PrintWriter outReportErrorWriter, String validationMessage) {
+		boolean valid = true;
+		if ((attributeValue.equalsIgnoreCase(KFSParameterKeyConstants.HYPHEN) || StringUtils.isBlank(attributeValue)) && isAttributeRequired) {
+			reportMsg += validationMessage.concat(attributeValue) + ") is required, ";
 			valid = false;
-			reportMsg += "BuildingAddressZipCode(" + archBuilding.getBuildingAddressZipCode() + ") is longer than " + maxBuildingAddressZipCodeLen + ", ";
 		}
-		if (isAttributeNotValid(isBuildingAddressCountryCodeRequired, archBuilding.getBuildingAddressCountryCode())) {
-			valid = false; 
-			reportMsg += "BuildingAddressCountyCode(" + archBuilding.getBuildingAddressCountryCode() + ") is Not Valid, ";
+
+		if (validationMessage.equalsIgnoreCase("BuildingAddressCountryCode(")) {
```

Comment:
Is this needed as the length was not checked in KFS3 for the Building Address Country Code?